AngularJS/src/org/angular2/entities/metadata/stubs/Angular2MetadataDirectiveStubBase.java

```diff
@@ -98,6 +133,60 @@ protected FlagsStructure getFlagsStructure() {
     return FLAGS_STRUCTURE;
   }
 
+  /**
+   * Given a JSON metadata document describing the Angular Directive, finds the {@code @Attribute}
+   * annotated constructor's parameters and creates a map to link the {@code @Attribute} alias
+   * to its parameter index.
+   *
+   * @param source
+   *       The JSON metadata document
+   */
+  @NotNull
+  private Map<String, Integer> loadAttributesMapping(@NotNull final JsonObject source) {
+    return StreamEx.ofNullable(getPropertyValue(source.findProperty(MEMBERS), JsonObject.class))
+                   .map(toPropertyValue(CONSTRUCTOR, JsonArray.class))
+                   .nonNull()
+                   .map(JsonArray::getValueList)
+                   .flatMap(StreamEx::of)
+                   .map(toCast(JsonObject.class))
+                   .map(toPropertyValue(PARAMETER_DECORATORS, JsonArray.class))
+                   .nonNull()
+                   .findFirst()
+                   .map(this::buildAttributesMapping)
+                   .orElse(emptyMap());
+  }
+
+  @NotNull
+  private Map<String, Integer> buildAttributesMapping(@NotNull final JsonArray paramDecorators) {
+    // Checks if the input object represents the @Attribute decorator
+    final Predicate<JsonObject> isAttributeDecorator = object -> {
+      final JsonObject expr = getPropertyValue(object.findProperty(EXPRESSION), JsonObject.class);
+      final String decoratorName = expr != null
+            ? readStringPropertyValue(expr.findProperty(REFERENCE_NAME))
+            : null;
+      return ATTRIBUTE_DEC.equals(decoratorName);
+    };
+
+    return EntryStream.of(paramDecorators.getValueList())
+                      .mapValues(toCast(JsonArray.class))
```

Comment:
`selectValues(JsonArray)`

Reply:
@piotrtomiak thanks! `select*` is a lot cleaner.